src/PhpDoc/Tag/PropertyTag.php

```diff
@@ -10,25 +10,44 @@ class PropertyTag
 
 	public function __construct(
 		private Type $type,
-		private bool $readable,
-		private bool $writable,
+		private ?Type $readableType,
+		private ?Type $writableType,
```

Comment:
One downside compared to #2294 is that sharing `PropertyTag` makes it impossible to e.g. inherit `@property` definition and then override `@property-write` in the subclass. (Though I did not bother implementing it yet.)